### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by elie-g**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for creating a TypeScript definition. Can you follow [this styleguide](https://github.com/sindresorhus/typescript-definition-style-guide)?

File: package.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "test": "xo && ava"
   },
   "files": [
-    "index.js"
+    "index.js",
+    "types/index.d.ts"
```
Comment (MrTarantula) [inline package.json]: To follow the guide, move this and other files from `types` folder to the root.

File: package.json [hunk 2]
```diff
@@ -38,6 +39,7 @@
   },
   "devDependencies": {
     "ava": "*",
+    "typescript": "^3.0.1",
```
Comment (MrTarantula) [inline package.json]: TS is not necessary for development. Remove this dependency and `tsconfig.json`.

File: types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+declare function show(stream?: WritableStream): void;
```
Comment (MrTarantula) [inline types/index.d.ts]: This file needs [TSDoc comments](https://github.com/sindresorhus/typescript-definition-style-guide#documentation).

**Comment by sindresorhus** [comment]
@DrunkenPoney Ping :)

**Comment by elie-g** [comment]
Done! Sorry for the delay

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Shows the cursor.
+ * 
+ * @param stream (Default: process.stderr)
```
Comment (sindresorhus) [inline index.d.ts:4]: There should be a `-` here. See the style guide. Also see the style guide regarding `Default` formatting.

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ */
+declare function show(stream?: WritableStream): void;
```
Comment (sindresorhus) [inline index.d.ts:6]: Import `WritableStream` explicitly instead of using the implicit global.

File: index.d.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+ * Hides the cursor.
+ * 
+ * @param stream (Default: process.stderr)
+ */
+declare function hide(stream?: WritableStream): void;
+
+/**
+ * Toggles the cursor visibility.
+ * 
+ * @param force Useful to show or hide the cursor based on a boolean
```
Comment (sindresorhus) [inline index.d.ts:18]: Description should end in a `.`

File: test.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import {show, hide, toggle} from '..';
+declare const process: {stdout: WritableStream};
```
Comment (sindresorhus) [inline test.ts:2]: This should be correctly imported and not declared.

File: index.d.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+ * @param stream (Default: process.stderr)
+ */
+declare function toggle(force?: boolean, stream?: WritableStream): void;
+
+export {show, hide, toggle};
```
Comment (sindresorhus) [inline index.d.ts:23]: The export should be a default export.

File: test.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import {show, hide, toggle} from '..';
```
Comment (sindresorhus) [inline test.ts:1]:
```suggestion
import cliCursor from '..';
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
> For packages with a default export, use export default foo;, not export = foo;. You will have to add module.exports.default = foo; to the index.js file.

> The pull request should have the title Add TypeScript definition. (Copy-paste it so you don't get it wrong)

**Comment by sindresorhus** [comment]
Ping

**Comment by sindresorhus** [comment]
Closing in favor of #5.

**Pull request closed without merging**